pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1504,24 +1521,38 @@ func (volTopology *wcpControllerVolumeTopology) GetSharedDatastoresInTopology(ct
 		// Call GetCandidateDatastores for each cluster moref. Ignore the vsanDirectDatastores for now.
 		if !isPodVMOnStretchedSupervisorEnabled {
 			// This code block assume we have 1 Cluster Per AZ
-			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMorefs[0], false)
+			getCandidateFunc := volTopology.getCandidateDatastoresFunc
+			if getCandidateFunc == nil {
+				getCandidateFunc = cnsvsphere.GetCandidateDatastoresInCluster
+			}
+			accessibleDs, _, err := getCandidateFunc(ctx, params.Vc, clusterMorefs[0], false)
 			if err != nil {
 				return nil, logger.LogNewErrorf(log,
 					"failed to find candidate datastores to place volume in cluster %q. Error: %v",
 					clusterMorefs[0], err)
 			}
 			params.TopoSegToDatastoresMap[zone] = accessibleDs
 			sharedDatastores = append(sharedDatastores, accessibleDs...)
+			log.Debugf("PodVMOnStretchedSupervisor Not Enabled. Topology: %+v, Zone: %s, "+
+				"Topology Segments to Datastore Map: %+v, Datastores: %v",
+				params.TopologyRequirement, zone, params.TopoSegToDatastoresMap, sharedDatastores)
 		} else {
 			// This code block adds support for multiple vSphere Clusters Per AZ
 			// sharedDatastores will be calculated for all clusters within AZ
-			sharedDatastoresForclusterMorefs, err := getSharedDatastoresInClusters(ctx, clusterMorefs, params.Vc)
+			getSharedFunc := volTopology.getSharedDatastoresInClustersFunc
+			if getSharedFunc == nil {
+				getSharedFunc = getSharedDatastoresInClusters
```

Comment:
same question as above, when do we set getSharedFunc = getSharedDatastoresInClusters

Reply:
I made some changes so it does not do nil check any more.